src/rat_king_parser/config_parser/rat_config_parser.py

```diff
@@ -171,11 +171,22 @@ def _decrypt_and_decode_config(
             )
         if self.remap_config:
             sorted_decoded_config = OrderedDict()
+            normalized_fields = []
             for k in sorted(config_fields_map.keys()):
                 key_name = config_fields_map[k]
                 value = decoded_config[key_name]
-                key_name, value = check_key_n_value(key_name, value)
-                sorted_decoded_config[key_name] = value
+                key_normalized, value = check_key_n_value(key_name, value)
+                if key_normalized != key_name:
+                    normalized_fields.append(key_name)
+                sorted_decoded_config[key_normalized] = value
+            # Ensure config items added by decryptors dynamically are preserved
+            sorted_decoded_config.update(
+                {
```

Comment:
the issue here is that we return for example list for hosts, ports, but then in the loop we restore old value, so we don't use "normalized" one

Reply:
`key not in config_fields_map` is {RVA:key} so key won't be in keys(RVAs), it could be in `.values()` but the proper value to check here is `sorted_decoded_config` with changing this, all works as expected

Reply:
pushed fix

Reply:
Ah good catch! That slipped by as the output was the same regardless, but you're right in that the check on `config_fields_map` was wrong - Nice!